### PR TITLE
Custom Op name for load_assets_from_dbt_*

### PR DESCRIPTION
**Pull request by wkeifenheim**
## Summary & Motivation
See issue  #13873

This PR does not include capturing other CLI parameters  [`profile`, `target`, `vars`] towards dynamically generating op names.

## How I Tested These Changes
The `test_op_custom_name` test builds two asset collections from dbt, each with a different target, and no other differences (of dbt config). Loading from the different manifests while (distinctly) setting the new `op_name` parameter yields `op` names that do not collide.

_Example of one custom op name from the dbt load, within the `AssetsDefintion` object_
image

I have also manually tested this by modifying the `assets_dbt_python` example to load multiple asset groups of the dbt project. Each group targets a different database destination, and in this example consume from a common source. The multiple instances of the dbt project worked as intended. Without the custom `op_name` field we would fail once the different dbt asset groups are added to `Definitions`.
image

### CODE REVIEW

**Comment by smackesey** [comment]
cc @OwenKephart

File: python_modules/libraries/dagster-dbt/dagster_dbt_tests/test_asset_defs.py [hunk 1]
```diff
@@ -655,6 +656,37 @@ def test_op_config(
     assert all_keys == expected_keys
 
 
+@pytest.mark.parametrize("instances", [[{"target": "target_a"}, {"target": "target_b"}]])
+def test_op_custom_name(instances, test_project_dir, dbt_config_dir, dbt_target_dir):
```
Comment (OwenKephart) [inline python_modules/libraries/dagster-dbt/dagster_dbt_tests/test_asset_defs.py:660]: I think this test could be simplified a ton, just by having all of the load_assets_from_dbt_manifest calls share the same manifest.json file (sample_manifest.json would work).

as long as the keys are distinct (which is handled by the key_prefix already), the actual contents of the manifest shouldn't matter for the purposes of this test.

also, as there's technically only one set of instances, we can probably skip parameterizing the test and just hard-code those in.

File: python_modules/libraries/dagster-dbt/dagster_dbt_tests/test_asset_defs.py [hunk 2]
```diff
@@ -655,6 +656,37 @@ def test_op_config(
...
+                    manifest_json=json.load(manifest_json),
+                    key_prefix=[instance['target'], "duckdb", "test-schema"],
+                    source_key_prefix=["duckdb"],
+                    op_name=f"{instance['target']}_dbt_op",
+                    node_info_to_group_fn=lambda ni: f"{instance['target']}",
+                    dbt_resource_key=f"dbt_{instance['target']}",
+                )
+            )
+    op_names = [asset_group.op.name for asset_group in dbt_assets]
+    assert len(op_names) == len(set(op_names)),\
```
Comment (OwenKephart) [inline python_modules/libraries/dagster-dbt/dagster_dbt_tests/test_asset_defs.py:684]: a bit of a nitpick, but would prefer multi-line strings to be in this format:

```
assert len(op_names) == len(set(op_names)), (
    f"Multiple instances of a dbt project cannot have the same op name.\n"
    f"dbt targets were: {instances}\n"
    f"op names generated were: {op_names}"
)
```

**Review by OwenKephart — CHANGES_REQUESTED** [review]
thanks for making this change! everything code-wise seems good to me, but had some comments around simplifying the test code / formatting.

also, make sure to run `make black` at the root of the repo before committing.

**Review by OwenKephart — APPROVED** [review]
Thanks again for the contribution! Merging this in now, and it'll be in next week's release

**Pull request merged**